### PR TITLE
[DEV-1369] Anchor Links with hash

### PR DESCRIPTION
**Pull request by jeremygordillo**
#### List of Changes

- Update link parser to manage hash inside links
- Add tests

#### Motivation and Context

We solve internal navigation with links that contains a hash.
For example: 
Before: https://developer.pagopa.it/app-io/guides/io-guida-tecnica/v5.0/setup-iniziale/#accordo-di-adesione
After: https://developer.pagopa.it/app-io/guides/io-guida-tecnica/v5.0/setup-iniziale#accordo-di-adesione

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .changeset/little-toes-rescue.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"gitbook-docs": patch
+---
+
+[DEV-1369] Fix links that contains a hash
```
Comment (marcobottaro) [inline .changeset/little-toes-rescue.md:5]: Did you mean links that ends with `/#`?

File: packages/gitbook-docs/src/__tests__/parseContent.test.ts [hunk 2]
```diff
@@ -149,6 +145,41 @@ describe('parseContent', () => {
     ]);
   });
 
+  it('should convert href as expected given an link with anchor', () => {
```
Comment (marcobottaro) [inline packages/gitbook-docs/src/__tests__/parseContent.test.ts:148]:
```suggestion
  it('should convert href as expected given a link with an anchor', () => {
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: packages/gitbook-docs/src/markdoc/schema/link.ts [hunk 3]
```diff
@@ -47,6 +47,15 @@ export const link: Schema = {
       : titleFromAnchor
       ? [titleFromAnchor]
       : childrenTreeNode;
+
+    if (attrs.href.match(/\/#/)) {
+      const splitted = attrs.href.split('/');
+      const anchor = splitted[splitted.length - 1].replace('/', '');
+      const href = splitted.slice(0, -1).join('/');
+      const linkAttrs = { ...attrs, href: `${href}${anchor}` };
+      return new Tag('Link', linkAttrs, children);
```
Comment (datalek) [inline packages/gitbook-docs/src/markdoc/schema/link.ts:56]: I would consider putting this login inside [LinkAttr](https://github.com/pagopa/developer-portal/blob/558d23c/packages/gitbook-docs/src/markdoc/attributes.ts#L27-L68)

Comment (jeremygordillo) [reply packages/gitbook-docs/src/markdoc/schema/link.ts]: I moved the logic into the [convertLink](https://github.com/pagopa/developer-portal/blob/558d23c/packages/gitbook-docs/src/markdoc/attributes.ts#L19) function .

File: .changeset/little-toes-rescue.md (hunk 1 above)
Comment (jeremygordillo) [reply .changeset/little-toes-rescue.md]: I updated changeset message

File: packages/gitbook-docs/src/markdoc/attributes.ts [hunk 4]
```diff
@@ -16,13 +16,23 @@ export class BooleanOrNullAttr {
...
+  if (replaced.match(/\/#/)) {
+    const splitted = replaced.split('/');
+    const anchor = splitted[splitted.length - 1].replace('/', '');
+    const url = splitted.slice(0, -1).join('/');
+    return `${url}${anchor}`;
+  }
+
+  return replaced;
+};
+
```
Comment (datalek) [inline packages/gitbook-docs/src/markdoc/attributes.ts:35]: I think the following implementation is equivalent:
```suggestion
const convertLink = (link: string): string => 
  link
    .replace('/README.md', '')
    .replace('README.md', '')
    .replace('.md', '')
    .replace(new RegExp('^(.*?)\\/?$'), '$1')
    // uniform links like 'this/is/a/path/#fragment' to 'this/is/a/path#fragment'
    .replace(new RegExp('/#'), '#');

```

File: .changeset/little-toes-rescue.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"gitbook-docs": patch
+---
+
+[DEV-1369] Fix anchor links
```
Comment (marcobottaro) [inline .changeset/little-toes-rescue.md:5]:
```suggestion
Fix anchor links parsing
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by marcobottaro — APPROVED** [review]

File: packages/gitbook-docs/src/__tests__/parseContent.test.ts [hunk 6]
```diff
@@ -119,13 +115,13 @@ describe('parseContent', () => {
   it('should preserve a single space into a td tag nested into other tags', () => {
     expect(
       parseContent(
-        '<td><strong>before space</strong> <a><strong>after space</strong></a></td>',
+        '<td><strong>before space</strong> <a href="target-link"><strong>after space</strong></a></td>',
```
Comment (datalek) [inline packages/gitbook-docs/src/__tests__/parseContent.test.ts:118]: Do you need this test? If you are testing anchor parsing then create a specific test

**Review by MarcoPonchia — APPROVED** [review]

**Review by datalek — APPROVED** [review]

**Pull request merged**